README.md

```diff
@@ -0,0 +1,560 @@
+
+# Sample Android Native Application
+
+## Introduction
+
+This is a sample Android Java application showing use of DataSync, [Keycloak](https://www.keycloak.org/about.html) and Unified Push using native upstream SDK's. Application is sending requests to [Ionic showcase server](https://github.com/aerogear/ionic-showcase/tree/master/server) which is a GraphQL server. 
+
+- For DataSync, application uses Apollo Client to query, mutate and subscribe. 
+- For authorization we are using [AppAuth](https://github.com/openid/AppAuth-Android) to connect
+ with Keycloak. 
+- For Unifiedpush support we are using Aerogear SDK.
+
+## Implementation
+### 1. DataSync
+#### Generating queries, mutations and subscriptions
+  To generate queries, mutations and subscriptions of running GraphQL server [Apollo Codegen](https://github.com/apollographql/apollo-tooling) was used.
+
```

Comment:
For people that haven't used Graphql before, it might be good to have some links here for them to read and understand how graphql works and the basic concepts.